<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from bones.urdf.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="bones">
  <!-- ros control plugin -->
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      <robotNamespace>/ruff</robotNamespace>
      <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>
    </plugin>
  </gazebo>
  <!-- adding imu sensor to the base link -->
  <!--IMU : joint , link , sensor  -->

  <link name="base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.02"/>
      <mass value="2.49007846050021"/>
      <inertia ixx="0.00489492059233199" ixy="-1.59495229293445E-11" ixz="-2.61866110672266E-14" iyy="0.00625744875770968" iyz="7.8598187747419E-15" izz="0.00969112831967179"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/base_link.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.7 0.7 0.9 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/base_link.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="FL1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.05 0.02 0"/>
      <mass value="0.35871"/>
      <inertia ixx="0.0003145" ixy="-1.6766E-06" ixz="-9.9068E-06" iyy="0.00044328" iyz="-2.576E-06" izz="0.00020521"/>
    </inertial>
    <visual>
      <origin rpy="0.0 0.0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/shoulder.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.3 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0.0 0.0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/shoulder.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="FL1_" type="revolute">
    <origin rpy="0 0 0" xyz="0.2498 0.1502 -0.0101"/>
    <parent link="base_link"/>
    <child link="FL1"/>
    <axis xyz="1 0 0"/>
    <limit
      lower="-0.52333"
      upper="0.52333"
      effort="200"
      velocity="10" />
  </joint>
  <link name="FL2">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.26 0 -0.05"/>
      <mass value="0.421019079143915"/>
      <inertia ixx="0.000307852923020911" ixy="3.19242906120105E-06" ixz="-6.61267552894505E-06" iyy="0.000852068112953311" iyz="2.25635506155653E-06" izz="0.00065120696734265"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/femur_left.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.3 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/femur_left.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="FL2_" type="revolute">
    <origin rpy="0 -0.7853981633974483 0" xyz="0.064 0.068 0.0"/>
    <parent link="FL1"/>
    <child link="FL2"/>
    <axis xyz="0 1 0"/>
    <limit
      lower="-0.52333"
      upper="0.52333"
      effort="200"
      velocity="10" />
  </joint>
  <link name="FL3">
    <inertial>
      <origin rpy="0 0 0" xyz="0.15 0 0"/>
      <mass value="0.155935206674882"/>
      <inertia ixx="0.000710580255126011" ixy="-4.6955016006765E-08" ixz="5.54725888406585E-08" iyy="0.0011552209842016" iyz="5.86784342700648E-08" izz="0.000459413413981537"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/tibia.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.3 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/tibia.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="FL3_" type="revolute">
    <origin rpy="0 1.5707963267948966 0" xyz="-0.1805 0.029 -0.059"/>
    <parent link="FL2"/>
    <child link="FL3"/>
    <axis xyz="0 1 0"/>
    <limit
      lower="-0.52333"
      upper="0.52333"
      effort="200"
      velocity="10" />
  </joint>
  <link name="FL3_f">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.002"/>
      <inertia ixx="0.00001" ixy="0" ixz="0" iyy="0.00001" iyz="0" izz="0.00001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.005"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.3 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.005"/>
      </geometry>
    </collision>
  </link>
  <joint name="FL3_f_" type="fixed">
    <origin rpy="0 -1.5707963267948966 0" xyz="0.29 0 0"/>
    <parent link="FL3"/>
    <child link="FL3_f"/>
  </joint>
  <link name="FR1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.05 0.02 0"/>
      <mass value="0.35871"/>
      <inertia ixx="0.0003145" ixy="-1.6766E-06" ixz="-9.9068E-06" iyy="0.00044328" iyz="-2.576E-06" izz="0.00020521"/>
    </inertial>
    <visual>
      <origin rpy="-3.141592653589793 0.0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/shoulder.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.3 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="-3.141592653589793 0.0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/shoulder.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="FR1_" type="revolute">
    <origin rpy="0 0 0" xyz="0.2498 -0.1502 -0.0101"/>
    <parent link="base_link"/>
    <child link="FR1"/>
    <axis xyz="1 0 0"/>
    <limit
      lower="-0.52333"
      upper="0.52333"
      effort="200"
      velocity="10" />
  </joint>
  <link name="FR2">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.26 0 -0.05"/>
      <mass value="0.421019079143915"/>
      <inertia ixx="0.000307852923020911" ixy="3.19242906120105E-06" ixz="-6.61267552894505E-06" iyy="0.000852068112953311" iyz="2.25635506155653E-06" izz="0.00065120696734265"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/femur_right.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.3 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/femur_right.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="FR2_" type="revolute">
    <origin rpy="0 -0.7853981633974483 0" xyz="0.064 -0.068 0.0"/>
    <parent link="FR1"/>
    <child link="FR2"/>
    <axis xyz="0 1 0"/>
    <limit
      lower="-0.52333"
      upper="0.52333"
      effort="200"
      velocity="10" />
  </joint>
  <link name="FR3">
    <inertial>
      <origin rpy="0 0 0" xyz="0.15 0 0"/>
      <mass value="0.155935206674882"/>
      <inertia ixx="0.000710580255126011" ixy="-4.6955016006765E-08" ixz="5.54725888406585E-08" iyy="0.0011552209842016" iyz="5.86784342700648E-08" izz="0.000459413413981537"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/tibia.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.3 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/tibia.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="FR3_" type="revolute">
    <origin rpy="0 1.5707963267948966 0" xyz="-0.1805 -0.029 -0.059"/>
    <parent link="FR2"/>
    <child link="FR3"/>
    <axis xyz="0 1 0"/>
    <limit
      lower="-0.52333"
      upper="0.52333"
      effort="200"
      velocity="10" />
  </joint>
  <link name="FR3_f">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.002"/>
      <inertia ixx="0.00001" ixy="0" ixz="0" iyy="0.00001" iyz="0" izz="0.00001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.005"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.3 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.005"/>
      </geometry>
    </collision>
  </link>
  <joint name="FR3_f_" type="fixed">
    <origin rpy="0 -1.5707963267948966 0" xyz="0.29 0 0"/>
    <parent link="FR3"/>
    <child link="FR3_f"/>
  </joint>
  <link name="RL1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.05 0.02 0"/>
      <mass value="0.35871"/>
      <inertia ixx="0.0003145" ixy="-1.6766E-06" ixz="-9.9068E-06" iyy="0.00044328" iyz="-2.576E-06" izz="0.00020521"/>
    </inertial>
    <visual>
      <origin rpy="0.0 -3.141592653589793 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/shoulder.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.3 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0.0 -3.141592653589793 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/shoulder.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="RL1_" type="revolute">
    <origin rpy="0 0 0" xyz="-0.2498 0.1502 -0.0101"/>
    <parent link="base_link"/>
    <child link="RL1"/>
    <axis xyz="1 0 0"/>
    <limit
      lower="-0.52333"
      upper="0.52333"
      effort="200"
      velocity="10" />
  </joint>
  <link name="RL2">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.26 0 -0.05"/>
      <mass value="0.421019079143915"/>
      <inertia ixx="0.000307852923020911" ixy="3.19242906120105E-06" ixz="-6.61267552894505E-06" iyy="0.000852068112953311" iyz="2.25635506155653E-06" izz="0.00065120696734265"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/femur_left.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.3 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/femur_left.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="RL2_" type="revolute">
    <origin rpy="0 -0.7853981633974483 0" xyz="-0.064 0.068 0.0"/>
    <parent link="RL1"/>
    <child link="RL2"/>
    <axis xyz="0 1 0"/>
    <limit
      lower="-0.52333"
      upper="0.52333"
      effort="200"
      velocity="10" />
  </joint>
  <link name="RL3">
    <inertial>
      <origin rpy="0 0 0" xyz="0.15 0 0"/>
      <mass value="0.155935206674882"/>
      <inertia ixx="0.000710580255126011" ixy="-4.6955016006765E-08" ixz="5.54725888406585E-08" iyy="0.0011552209842016" iyz="5.86784342700648E-08" izz="0.000459413413981537"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/tibia.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.3 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/tibia.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="RL3_" type="revolute">
    <origin rpy="0 1.5707963267948966 0" xyz="-0.1805 0.029 -0.059"/>
    <parent link="RL2"/>
    <child link="RL3"/>
    <axis xyz="0 1 0"/>
    <limit
      lower="-0.52333"
      upper="0.52333"
      effort="200"
      velocity="10" />
  </joint>
  <link name="RL3_f">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.002"/>
      <inertia ixx="0.00001" ixy="0" ixz="0" iyy="0.00001" iyz="0" izz="0.00001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.005"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.3 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.005"/>
      </geometry>
    </collision>
  </link>
  <joint name="RL3_f_" type="fixed">
    <origin rpy="0 -1.5707963267948966 0" xyz="0.29 0 0"/>
    <parent link="RL3"/>
    <child link="RL3_f"/>
  </joint>
  <link name="RR1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.05 0.02 0"/>
      <mass value="0.35871"/>
      <inertia ixx="0.0003145" ixy="-1.6766E-06" ixz="-9.9068E-06" iyy="0.00044328" iyz="-2.576E-06" izz="0.00020521"/>
    </inertial>
    <visual>
      <origin rpy="-3.141592653589793 -3.141592653589793 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/shoulder.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.3 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="-3.141592653589793 -3.141592653589793 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/shoulder.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="RR1_" type="revolute">
    <origin rpy="0 0 0" xyz="-0.2498 -0.1502 -0.0101"/>
    <parent link="base_link"/>
    <child link="RR1"/>
    <axis xyz="1 0 0"/>
    <limit
      lower="-0.52333"
      upper="0.52333"
      effort="200"
      velocity="10" />
  </joint>
  <link name="RR2">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.26 0 -0.05"/>
      <mass value="0.421019079143915"/>
      <inertia ixx="0.000307852923020911" ixy="3.19242906120105E-06" ixz="-6.61267552894505E-06" iyy="0.000852068112953311" iyz="2.25635506155653E-06" izz="0.00065120696734265"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/femur_right.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.3 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/femur_right.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="RR2_" type="revolute">
    <origin rpy="0 -0.7853981633974483 0" xyz="-0.064 -0.068 0.0"/>
    <parent link="RR1"/>
    <child link="RR2"/>
    <axis xyz="0 1 0"/>
    <limit
      lower="-0.52333"
      upper="0.52333"
      effort="200"
      velocity="10" />
  </joint>
  <link name="RR3">
    <inertial>
      <origin rpy="0 0 0" xyz="0.15 0 0"/>
      <mass value="0.155935206674882"/>
      <inertia ixx="0.000710580255126011" ixy="-4.6955016006765E-08" ixz="5.54725888406585E-08" iyy="0.0011552209842016" iyz="5.86784342700648E-08" izz="0.000459413413981537"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/tibia.stl"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.3 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="../meshes/tibia.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="RR3_" type="revolute">
    <origin rpy="0 1.5707963267948966 0" xyz="-0.1805 -0.029 -0.059"/>
    <parent link="RR2"/>
    <child link="RR3"/>
    <axis xyz="0 1 0"/>
    <limit
      lower="-0.52333"
      upper="0.52333"
      effort="200"
      velocity="10" />
  </joint>
  <link name="RR3_f">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.002"/>
      <inertia ixx="0.00001" ixy="0" ixz="0" iyy="0.00001" iyz="0" izz="0.00001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.005"/>
      </geometry>
      <material name="">
        <color rgba="0.2 0.2 0.3 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.005"/>
      </geometry>
    </collision>
  </link>
  <joint name="RR3_f_" type="fixed">
    <origin rpy="0 -1.5707963267948966 0" xyz="0.29 0 0"/>
    <parent link="RR3"/>
    <child link="RR3_f"/>
  </joint>
  <!-- Rear right  -->
  <!-- Transmission for the joints -->
  <transmission name="FL1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="FL1_motor">
      <mechanicalReduction>1</mechanicalReduction>
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </actuator>
    <joint name="FL1_">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
  </transmission>
  <transmission name="FL2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="FL2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="FL2_">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
  </transmission>
  <transmission name="FL3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="FL3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="FL3_">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
  </transmission>
  <transmission name="FR1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="FR1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="FR1_">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
  </transmission>
  <transmission name="FR2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="FR2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="FR2_">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
  </transmission>
  <transmission name="FR3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="FR3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="FR3_">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
  </transmission>
  <transmission name="RL1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="RL1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="RL1_">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
  </transmission>
  <transmission name="RL2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="RL2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="RL2_">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
  </transmission>
  <transmission name="RL3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="RL3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="RL3_">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
  </transmission>
  <transmission name="RR1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="RR1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="RR1_">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
  </transmission>
  <transmission name="RR2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="RR2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="RR2_">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
  </transmission>
  <transmission name="RR3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="RR3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="RR3_">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
  </transmission>
</robot>
